exporter/fileexporter/config.go

```diff
@@ -79,3 +85,34 @@ func (cfg *Config) Validate() error {
 	}
 	return nil
 }
+
+// Unmarshal a confmap.Conf into the config struct.
+func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
+	if componentParser == nil {
+		return errors.New("empty config for file exporter")
+	}
+	// first load the config normally
+	err := componentParser.UnmarshalExact(cfg)
+	if err != nil {
+		return err
+	}
+
+	// next manually search for protocols in the confmap.Conf,
+	// if rotation is not present it means it is disabled.
+	if !componentParser.IsSet(rotationFieldName) {
+		return nil
+	}
+	rotation := componentParser.Get(rotationFieldName)
+	if rotation == nil {
+		cfg.Rotation = &Rotation{MaxBackups: defaultMaxBackups}
+		return nil
+	}
+	rotationCfg, err := componentParser.Sub(rotationFieldName)
```

Comment:
I think what we want to do from here is more like this:

```go
	rotationConfmap, err := componentParser.Sub(rotationFieldName)
	rotationCfg := newDefaultRotationConfig()
	rotationConfmap.UnmarshalExact(rotationCfg)
	cfg.Rotation  = rotationCfg
```

The idea is that we typically want to instantiate the default config struct, and then unmarshal on top of it. This ensures that default settings apply unless they are overwritten.